### DISCUSSION TITLE
Q: How to implement Iced in a custom game engine (DirectX)

### DISCUSSION THREAD

**Discussion by cxcubehd** [hecrj/iced#1690] [Q&A]
Hello, 
I would like to ask what I would need to do in order to implement Iced into my custom game engine running on DirectX? Note that the window handling stuff is already implemented and I don't want to change that. What I would like to want is to send all input events manually and render the UI on a separate texture. Any help / advice would be appreciated!

**Comment by manokara** [comment]
You could look [here](https://github.com/iced-rs/iced/blob/master/winit/src/application.rs#L339) (going down to the `RedrawRequested` event) for windowing integration and [the iced-wgpu crate](https://github.com/iced-rs/iced/tree/master/wgpu/src) (starting from `backend.rs` and `window/compositor.rs`) for the rendering part. wgpu is the reference renderer, but if you'd rather have OpenGL as a point of reference you could look at the same files in `iced-glutin` (windowing) and `iced-glow` (rendering), but they're not as feature complete or optmized.

**Reply by thunderstorm010** [reply]
And they will be removed in the future, as well.

**Comment by trace7087** [comment]
hello there,

im hoping someone can help me please?
im using parallels 18 on my mac m2 with a windows 11, i wanted to play the new Hogwarts on windows (virtual machine) but it wont even load, keeps saying about it needed directx installed.
ive tried everything.

**Reply by thunderstorm010** [reply]
Relation of this question to iced?
